files/ruby_plugin_helper.rb

```diff
@@ -0,0 +1,48 @@
+TARGET_OPTS = %i[uri name config].freeze
+
+def config_lookup(opts)
+  template = opts.select { |k, _| TARGET_OPTS.include?(k) }
+  lookups = []
+  walk_vals(template) do |value|
+    # First collect the lookups to make
+    lookups << value if value.is_a?(String)
+    value
+  end
+  lookups.uniq!
+  lookups.map! { |lookup| lookup.split('.') }
+ end
+
+# 'template' parameter is the value of `target_mapping` from the inventory
+# 'lookups' is an array of hashes, where each hash is the lookup for a target
+# Returns an array of hashes to be sent to the inventory
+def create_inventory(opts, lookups)
+  template = opts.select { |k, _| TARGET_OPTS.include?(k) }
```

Comment:
We should just take the template as an argument. Assuming we know where it will be will limit the uses for this later.

I think we should leave the mapping up to the caller as well. 

Something like `apply_mapping(mapping, data)` seems a lot easier for module authors to understand and is more flexible for different use cases.